File: 70-climbing-stairs/climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if(n <= 2){
        return n;
    }
    let one = 1;
    let two = 2;
    let curr;
    for(let x = 3; x <= n; x++){
        curr = one + two;
        one = two;
        two = curr;
    }
    return curr;
};

File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if(n == 2):
            return 2
        if(n == 1):
            return 1

        dp = []
        dp.append(1)
        dp.append(2)

        for x in range(2, n):
            dp.append(dp[x-2] + dp[x-1])
        
        return dp[-1]